### DISCUSSION TITLE
How to set client ID in vsomeip or in CommonAPI SOMEIP

### DISCUSSION THREAD

**Discussion by NagarjunaReddyG5262** [COVESA/vsomeip#742] [Q&A]
Hi,

I want to set the client ID, while calling a method (Request and Response) either using vsomeip or CommonAPI for SOMEIP. Is it possible?
Does the client ID specified in Security configuration corresponds to the client ID(2bytes) of the someip frame format? Because I can see that the client ID is 0x0000 always even the application ID is given a non zero number.

**Comment by srl100** [comment]
Are you talking about the "const std::shared_ptr<CommonAPI::ClientId> _client" parameter that is passed to each of the stub methods? If so, I've got the same query (under discussion #712) - note that this one, I think, is (or should be?) the IP address of the client, not the 2 byte client ID passed in the SOME/IP frame.
